seaborn/categorical.py

```diff
@@ -1498,24 +1498,52 @@ def estimate_statistic(self, estimator, ci, n_boot):
             self.value_label = "{}({})".format(estimator.__name__,
                                                self.value_label)
 
-    def draw_confints(self, ax, at_group, confint, colors, **kws):
-
-        kws.setdefault("lw", mpl.rcParams["lines.linewidth"] * 1.8)
+    def draw_confints(self,
+                      ax, at_group,
+                      confint,
+                      colors,
+                      conf_lw=None,
+                      capsize=None,
+                      **kws):
+
+        if conf_lw:
+            kws.setdefault("lw", mpl.rcParams["lines.linewidth"] * conf_lw)
+        else:
+            kws.setdefault("lw", mpl.rcParams["lines.linewidth"] * 1.8)
 
-        for at, (ci_low, ci_high), color in zip(at_group,
-                                                confint,
-                                                colors):
-            if self.orient == "v":
-                ax.plot([at, at], [ci_low, ci_high], color=color, **kws)
-            else:
-                ax.plot([ci_low, ci_high], [at, at], color=color, **kws)
+        if capsize:
+            for at, (ci_low, ci_high), color in zip(at_group,
+                                                    confint,
+                                                    colors):
+                if self.orient == "v":
+                    ax.plot([at, at], [ci_low, ci_high], color=color, **kws)
+                    ax.plot([at - capsize / 2, at + capsize / 2],
+                            [ci_low, ci_low], color=color, **kws)
+                    ax.plot([at - capsize / 2, at + capsize / 2],
+                            [ci_high, ci_high], color=color, **kws)
+                else:
+                    ax.plot([ci_low, ci_high], [at, at], color=color, **kws)
+                    ax.plot([ci_low, ci_low],
+                            [at - capsize / 2, at + capsize / 2],
+                            color=color, **kws)
+                    ax.plot([ci_high, ci_high],
+                            [at - capsize / 2, at + capsize / 2],
+                            color=color, **kws)
+        else:
```

Comment:
This else clause is duplicating code. I would restructure to write the loop logic once and handle the capsize logic inside the loop.